### PR TITLE
Add on counter attack ability on Celestial Phantom

### PR DESCRIPTION
**Pull request by Polynucleotide**
From what I tested, Celestial Phantom does not kill ranged units any more and it kills Worldcore Golem.

### CODE REVIEW

File: Server/resources/accounts/duelyst.account.json [hunk 1]
```diff
@@ -1 +1,8202 @@
...,"hybrid":true}},"availabilityType":{"onPut":true}}],"manaCost":2,"price":100,"remainingNumber":20},{"name":"Tempest","description":"Deal 2 Damage to ALL characters","cardId":"duelyst_Tempest_2","spriteName":"icon_f1_tempest","type":"SPELL","spells":[{"spellId":"deal_2_dmg_everything","action":{"hpChange":-2,"durable":true,"duration":1},"target":{"dimensions":{"row":10,"column":18},"owner":{"own":true,"enemy":true},"cardType":{"hero":true,"minion":true,"player":true},"attackType":{"melee":true,"ranged":true,"hybrid":true}},"availabilityType":{"onPut":true}}],"manaCost":2,"price":100,"remainingNumber":20},{"name":"Tempest","description":"Deal 2 Damage to ALL characters","cardId":"duelyst_Tempest_3","spriteName":"icon_f1_tempest","type":"SPELL","spells":[{"spellId":"deal_2_dmg_everything","action":{"hpChange":-2,"durable":true,"duration":1},"target":{"dimensions":{"row":10,"column":18},"owner":{"own":true,"enemy":true},"cardType":{"hero":true,"minion":true,"player":true},"attackType":{"melee":true,"ranged":true,"hybrid":true}},"availabilityType":{"onPut":true}}],"manaCost":2,"price":100,"remainingNumber":20},{"name":"Void Pulse","description":"Deal 2 damage to the enemy General. Your
```
Comment (QLaille) [inline Server/resources/accounts/duelyst.account.json]: Was this intended ?

File: Server/src/server/gameCenter/models/game/Game.java [hunk 2]
```diff
@@ -558,6 +558,16 @@ private void applyOnAttackSpells(ServerTroop attackerTroop, ServerTroop defender
         }
     }
 
+    private void applyOnCounterAttackSpells(ServerTroop attacker, ServerTroop counterAttacker) {
```
Comment (QLaille) [inline Server/src/server/gameCenter/models/game/Game.java]: Can you move this function under all the lines in which it is used ? This is about clarifying the code, and jumping up and down doesn't help.

File: Server/resources/accounts/duelyst.account.json (hunk 1 above)
Comment (Polynucleotide) [reply Server/resources/accounts/duelyst.account.json]: No, I think I messed it up.

File: Server/src/server/gameCenter/models/game/Game.java (hunk 2 above)
Comment (Polynucleotide) [reply Server/src/server/gameCenter/models/game/Game.java]: Done

File: Server/resources/accounts/duelyst.account.json (hunk 1 above)
Comment (Polynucleotide) [reply Server/resources/accounts/duelyst.account.json]: I replaced this file with the master branch

**Pull request merged**